File: CPPScript/Получение ссылки на медиа-ресурс/Функция получения ссылки для torrent файлов.cpp
// ---- Формирование ссылки для просмотра torrent файла -----------------------
void GetTorrentMediaResourceLink(String sFile, int nIndex = 1) {
  String sDownloadsDir, sVal, sLink;

  sDownloadsDir = HmsTranscodingTempDirectory + "Torrents";
  sLink = Format('torrent:%s?index=%d&savepath=%s', [sFile, nIndex, sDownloadsDir]);
  if (HmsRegExMatch('--portbegin=(\\d+)', mpPodcastParameters, sVal)) sLink += '&portbegin='+sVal;
  if (HmsRegExMatch('--portend=(\\d+)'  , mpPodcastParameters, sVal)) sLink += '&portend='  +sVal;
  if (HmsRegExMatch('--sslport=(\\d+)'  , mpPodcastParameters, sVal)) sLink += '&sslport='  +sVal;
  MediaResourceLink = sLink;        
}

File: PascalScript/Получение ссылки на медиа-ресурс/Функция получения ссылки для torrent файлов.pas
// ---- Формирование ссылки для просмотра torrent файла -----------------------
Procedure GetTorrentMediaResourceLink(sFile: String; nIndex: Integer = 1);
Var
  sDownloadsDir, sVal, sLink: String;
Begin
  sDownloadsDir := HmsTranscodingTempDirectory + "Torrents";
  sLink := Format('torrent:%s?index=%d&savepath=%s', [sFile, nIndex, sDownloadsDir]);
  if HmsRegExMatch('--portbegin=(\\d+)', mpPodcastParameters, sVal) Then sLink := sLink+'&portbegin='+sVal;
  if HmsRegExMatch('--portend=(\\d+)'  , mpPodcastParameters, sVal) Then sLink := sLink+'&portend='  +sVal;
  if HmsRegExMatch('--sslport=(\\d+)'  , mpPodcastParameters, sVal) Then sLink := sLink+'&sslport='  +sVal;
  MediaResourceLink := sLink;        
End;
